### PR TITLE
Announce Go client & zbctl deprecation

### PR DESCRIPTION
**Pull request by megglos**
## Description

This is the announcement of the go client deprecation with 8.6.
See https://github.com/camunda/camunda/issues/21817.

PDP epic https://github.com/camunda/product-hub/issues/2438

Also relates to https://github.com/camunda/camunda-docs/pull/4228 (announces it)

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [x] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

**Review by Zelldon — DISMISSED** [review]

**Comment by mesellings** [comment]
@megglos I will review this for you now

File: docs/apis-tools/go-client/index.md [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
 id: index
-title: "Go client"
+title: "Go client (Deprecated)"
 sidebar_label: "Quick reference"
 description: "Instantiate the client by passing in the address of the cluster you want to connect to in a Go application to interact with Camunda 8."
 ---
 
+:::caution
+
+The Zeebe Go client will be deprecated as announced [here](../../reference/announcements/#deprecation-zeebe-go-client--zbctl)
```
Comment (mesellings) [inline docs/apis-tools/go-client/index.md:10]:
```suggestion
The Zeebe Go client will be officially deprecated with the 8.6 release. For more details, see [announcements](../../reference/announcements/#deprecation-zeebe-go-client--zbctl).
```

File: docs/reference/announcements.md [hunk 2]
```diff
@@ -50,6 +50,10 @@ The Zeebe Java client will not be developed further and will only receive bug fi
 - **Artifact ID change**:
   - The `artifactId` will change from `zeebe-client-java` to `camunda-client-java`.
 
+### Deprecation: Zeebe Go client & zbctl
+
+The Zeebe Go Client and zbctl will be officially deprecated with 8.6 release as part of our efforts to streamline the Camunda 8 API experience. This client and cli utility will not get released with Camunda 8.6, thus no longer receive new features and will be transitioned to a community-maintained status.
```
Comment (mesellings) [inline docs/reference/announcements.md:55]:
```suggestion
The Zeebe Go Client and zbctl will be officially deprecated with the 8.6 release as part of our efforts to streamline the Camunda 8 API experience. This client and cli utility will not get released with Camunda 8.6, will no longer receive new features, and will be transitioned to a community-maintained status.
```

File: docs/apis-tools/go-client/index.md [hunk 3]
```diff
@@ -1,10 +1,16 @@
 ---
 id: index
-title: "Go client"
+title: "Go client (Deprecated)"
 sidebar_label: "Quick reference"
 description: "Instantiate the client by passing in the address of the cluster you want to connect to in a Go application to interact with Camunda 8."
 ---
 
+:::caution
```
Comment (mesellings) [inline docs/apis-tools/go-client/index.md:8]: Should this caution go at the top of each page in this Go section? Users might not come to this first page, and could miss the warning if this is the case?

**Review by mesellings — CHANGES_REQUESTED** [review]
Hi @megglos I've made some suggestions which I think just border on being blocking and should be addressed please?

**Comment by akeller** [comment]
Can we follow the same pattern we do in the APIs and move the deprecated clients into a dedicated section in the sidebar? Also is deprecation and transfer to community-maintained happening at the same time? I don't see any feedback on the docs PoC PRs related to this topic. Will follow up on [Slack](https://camunda.slack.com/archives/C07L66RECJZ) and bring @mesellings into the conversation.

File: docs/apis-tools/go-client/index.md (hunk 3 above)
Comment (megglos) [reply docs/apis-tools/go-client/index.md:8]: good point, will do that!

Comment (megglos) [reply docs/apis-tools/go-client/index.md:8]: https://github.com/camunda/camunda-docs/pull/4264/commits/a5a4123c6a766ce4c6a50aa22c23c911668038a8

**Review by mesellings — APPROVED** [review]
Lgtm 👍

File: docs/reference/announcements.md [hunk 4]
```diff
@@ -50,6 +50,10 @@ The Zeebe Java client will not be developed further and will only receive bug fi
 - **Artifact ID change**:
   - The `artifactId` will change from `zeebe-client-java` to `camunda-client-java`.
 
+### Deprecation: Zeebe Go client & zbctl
+
+The Zeebe Go Client and zbctl will be officially deprecated with the 8.6 release as part of our efforts to streamline the Camunda 8 API experience. This client and cli utility will not get released with Camunda 8.6, will no longer receive new features, and will be transitioned to a community-maintained status.
```
Comment (aleksander-dytko) [inline docs/reference/announcements.md:55]:
```suggestion
The Zeebe Go Client and zbctl will be officially deprecated with the 8.6 release as part of our efforts to streamline the Camunda 8 API experience. This client and CLI utility will not get released starting with Camunda 8.6, will no longer receive new features, and will be transitioned to a community-maintained status. 
```

**Review by aleksander-dytko — DISMISSED** [review]
Pre-approved - just one suggestion to indicate that with >8.6 release Go Client won't be there

File: docs/reference/announcements.md (hunk 4 above)
Comment (megglos) [reply docs/reference/announcements.md]: @mesellings are you fine with this as well?

Comment (mesellings) [reply docs/reference/announcements.md]: Yep, it's a good clarification - well spotted @aleksander-dytko

Comment (mesellings) [reply docs/reference/announcements.md]: Or we could reword to "will no longer get released"

Comment (mesellings) [reply docs/reference/announcements.md]: But either is fine.

**Review by mesellings — APPROVED** [review]
Lgtm 👍

File: docs/apis-tools/go-client/go-get-started.md [hunk 5]
```diff
@@ -7,6 +7,12 @@ sidebar_label: "Getting started with the Go client"
 import Tabs from "@theme/Tabs";
 import TabItem from "@theme/TabItem";
 
+:::caution
```
Comment (christinaausley) [inline docs/apis-tools/go-client/go-get-started.md:10]: Should this perhaps be added to the current `8.5` docs, and should we remove all references of the Go client in the `next`, `8.6` docs?

Comment (christinaausley) [reply docs/apis-tools/go-client/go-get-started.md:10]: CC @megglos

**Pull request merged**